src/coreclr/src/tools/r2rtest/CpaotRunner.cs

```diff
@@ -94,20 +111,34 @@ protected override IEnumerable<string> BuildCommandLineArguments(IEnumerable<str
                 }
             }
 
-            StringComparer pathComparer = (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal);
-            HashSet<string> uniqueFolders = new HashSet<string>(pathComparer);
-
-            foreach (string assemblyFileName in assemblyFileNames)
+            if (_referenceFiles.Count == 0)
             {
-                uniqueFolders.Add(Path.GetDirectoryName(assemblyFileName));
-            }
+                // Use reference folders and find the managed assemblies we want to reference from them.
+                // This is the standard path when we want to compile folders and compare crossgen1 and crossgen2.
+                StringComparer pathComparer = (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal);
```

Comment:
Nit - might be worth a common helper, something like <code>OSPathCaseComparer</code>.

Reply:
Good idea